tests/utils.py

```diff
@@ -0,0 +1,44 @@
+import os
+import time
+from qiskit.circuit.random import random_circuit
+from qiskit.qasm2 import dumps as dump_qasm_2
+
+
+def build_random_qasm(
```

Comment:
Thanks for the changes! But we still want to have the gate set be a parameter, just not as a string like it was before. You can take in a list of gates (list of strings) and then transpile the circuit using qiskit `transpile` with optimization level 0.